### DISCUSSION TITLE
Provide timeout option for runs in Cy Dashboard

### DISCUSSION THREAD

**Discussion by abhidp** [cypress-io/cypress#16615] [Cypress Cloud]
Sometimes a run continues to run indefinitely until manually cancelled. 
This is because some specs keep on running till eternity for reasons unknown. 
As a result, the entire CI deployment pipeline is blocked.

As you can see from the screenshots, a spec has been running since 1+ Hour (which usually completes under 2 minutes otherwise) while all other specs have already completed running within 5 minutes combined (`parallelism: 10`). 
There are  no logs or anything to look into why this is still in progress.

Please provide a configurable option for Cy Dashboard to be able to cancel a run automatically if a run goes beyond a specified time limit a.k.a `timeout`

![specificDuration](https://user-images.githubusercontent.com/30851622/119094097-eb707600-ba53-11eb-904f-3bd5cbeb4d57.png)
<img width="1723" alt="allruns" src="https://user-images.githubusercontent.com/30851622/119094104-edd2d000-ba53-11eb-9d64-24a6c4b40ba0.png">

**Comment by abhidp** [comment]
anyone looking into this?

**Comment by abhidp** [comment]
another one, this has been happening consistently
![image](https://user-images.githubusercontent.com/30851622/120410774-0073d000-c397-11eb-81ff-d196f43001f2.png)

**Reply by jennifer-shehane** [reply]
The run will automatically cancel after 90 minutes of receiving no signal. I'll pass this information along to the Dashboard team for consideration.

**Reply by abhidp** [reply]
It keeps on happening on a regular basis now and our builds are badly affected. We cannot wait 90minutes for Cy dashboard to come back with an answer. Please provide a user configurable timeout value instead of a hardcoded default value of 90 minutes.

**Comment by abhidp** [comment]
![image](https://user-images.githubusercontent.com/30851622/124683928-809ad180-df11-11eb-9b20-ba2ea85fb07d.png)


![image](https://user-images.githubusercontent.com/30851622/124684118-ceafd500-df11-11eb-8cc5-fec15c86be10.png)

**Comment by nazad23** [comment]
Any updates on this one? I want to limit all builds to 1 hr max

**Comment by cheersli** [comment]
any update for set  timeout?
